src/main/java/it/pagopa/transactions/domain/TransactionWithRequestedAuthorization.java

```diff
@@ -0,0 +1,28 @@
+package it.pagopa.transactions.domain;
+
+import it.pagopa.transactions.annotations.AggregateRoot;
+import it.pagopa.transactions.documents.TransactionAuthorizationRequestedEvent;
+import it.pagopa.transactions.documents.TransactionAuthorizationStatusUpdatedEvent;
+import it.pagopa.transactions.domain.pojos.BaseTransaction;
+import it.pagopa.transactions.domain.pojos.BaseTransactionWithRequestedAuthorization;
+
+@AggregateRoot
+public final class TransactionWithRequestedAuthorization extends BaseTransactionWithRequestedAuthorization implements EventUpdatable<TransactionWithCompletedAuthorization, TransactionAuthorizationStatusUpdatedEvent>, Transaction {
+    TransactionWithRequestedAuthorization(BaseTransaction transaction, TransactionAuthorizationRequestedEvent event) {
+        super(transaction, event.getData());
+    }
+
+    @Override
+    public TransactionWithCompletedAuthorization apply(TransactionAuthorizationStatusUpdatedEvent event) {
```

Comment:
is new aggregate?

Reply:
Yes, each of `TransactionWith*` is a new aggregate. An alternative would have been to make the aggregate class generic and make it hold a `T data` field, like we do with events, but I felt separate classes were cleaner here. 🙂